tests/integration/test_pagure.py

```diff
@@ -19,54 +19,79 @@
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
-
 import os
 import subprocess
+from datetime import datetime
 from pathlib import Path
 
-import pytest
 from ogr.services.pagure import PagureService
+from vcr import use_cassette
 
 from tests.spellbook import git_set_user_email
 
 
-@pytest.mark.skipif(
-    condition=True, reason="Don't interact with a real pagure instance by default"
-)
+RESPONSE_HEADERS_TO_DROP = [
+    "X-Fedora-RequestID",
+    "content-security-policy",
+    "set-cookie",
+]
+
+
+def drop_sensitive_response_data(response):
+    for h in RESPONSE_HEADERS_TO_DROP:
+        try:
+            del response["headers"][h]
+        except KeyError:
+            pass
+    return response
+
+
 def test_basic_distgit_workflow(tmpdir):
-    pagure_token = os.getenv("PAGURE_TOKEN")
+    # if pagure_token is not set and we're recording, we'll get a token failure,
+    # but that's expected so no need to take care of it
+    with use_cassette(
+        path=str(Path(__file__).parent / "test_basic_distgit_workflow"),
+        filter_headers=["Authorization", "Cookie"],
+        before_record_response=drop_sensitive_response_data,
+    ) as cassette:
+
+        pagure_token = os.getenv("PAGURE_TOKEN")
 
-    pag = PagureService(
-        token=pagure_token,
-        repo="tmux-top",
-        namespace="rpms",
-        instance_url="https://src.stg.fedoraproject.org/",
-    )
+        pag = PagureService(
+            token=pagure_token, instance_url="https://src.stg.fedoraproject.org/",
+        )
 
-    print(pag.pagure.whoami())
+        assert pag.user.get_username()  # make sure the token is set
 
-    proj = pag.get_project()
+        proj = pag.get_project(repo="tmux-top", namespace="rpms",)
 
-    fork = proj.get_fork(create=True)
-    clone_url = fork.get_git_urls()["ssh"]
+        fork = proj.get_fork(create=True)
+        clone_url = fork.get_git_urls()["ssh"]
 
-    t = Path(tmpdir)
-    repo = t.joinpath("repo")
+        t = Path(tmpdir)
+        repo = t.joinpath("repo")
 
-    branch_name = "cookie"
+        branch_name = "cookie"
 
-    subprocess.check_call(["git", "clone", clone_url, repo])
-    git_set_user_email(repo)
-    subprocess.check_call(["git", "checkout", "-B", branch_name], cwd=repo)
+        subprocess.check_call(["git", "clone", clone_url, repo])
```

Comment:
It is sad that VCRpy does not work with `git`.

Reply:
clearly, git has nothing to do with python-requests